src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -29,43 +33,132 @@ public static IServiceCollection AddBlobDataStore(this IServiceCollection servic
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null"/>.</exception>
+        public static IServiceCollection AddBlobServiceClient(this IServiceCollection services)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
 
-            services.Add<BlobClientReadWriteTestProvider>()
-                .Singleton()
-                .AsService<IBlobClientTestProvider>();
+            services.AddOptions();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<BlobDataStoreConfiguration>>(new DefaultBlobDataStoreConfiguration()));
+            services.TryAddSingleton(p => BlobClientFactory.Create(p.GetRequiredService<IOptions<BlobDataStoreConfiguration>>().Value));
 
-            services.Add<BlobClientInitializer>()
-                .Singleton()
-                .AsService<IBlobClientInitializer>();
+            return services;
+        }
+
+        /// <summary>
+        ///  Adds a singleton <see cref="BlobServiceClient"/> to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <param name="configure">A delegate for configuring the <see cref="BlobDataStoreConfiguration"/>.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> or <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public static IServiceCollection AddBlobServiceClient(this IServiceCollection services, Action<BlobDataStoreConfiguration> configure)
+        {
+            EnsureArg.IsNotNull(configure, nameof(configure));
 
+            return services
+                .AddBlobServiceClient()
+                .Configure(configure);
+        }
+
+        /// <summary>
+        /// Adds a singleton <see cref="BlobServiceClient"/> to the specified <see cref="IServiceCollection"/> whose container
+        /// is already initialized by the time service can be resolved.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null"/>.</exception>
+        public static IServiceCollection AddInitializedBlobServiceClient(this IServiceCollection services)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
+
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<BlobDataStoreConfiguration>>(new DefaultBlobDataStoreConfiguration()));
```

Comment:
could we use [COnfigureOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.options.configureoptions-1?view=dotnet-plat-ext-5.0) to simplify this?

Reply:
I think that comes down to whether want to automatically read the configuration by default and bind the options to it. My inclination is "No," and for clients to perform the binding themselves based on their own configs. Then to that end we don't want all of the `IConfigureOptions` classes to register automatically.

Reply:
I'm thinking if we registered multiple implementation of `IConfigureOptions<BlobDataStoreConfiguration>`, which one will be used.

Reply:
They'll be executed sequentially based on the order they were added to the service collection